### PR TITLE
Add groovy expression

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: docs/modules/commons/pages/expressions.adoc [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+==== *_Info_*
+
+Executes groovy script and converts result to a string.
+
+==== *_Wording_*
+
+[source,gherkin]
+----
+#{groovy(script)}
```
Comment (valfirst) [inline docs/modules/commons/pages/expressions.adoc:17]: `groovy` -> `evalGroovy`

File: vividus-bdd-engine/src/main/resources/org/vividus/bdd/spring.xml [hunk 2]
```diff
@@ -113,5 +113,4 @@
     </bean>
 
     <bean id="parameterControls" class="org.jbehave.core.steps.ParameterControls" />
-
```
Comment (valfirst) [inline vividus-bdd-engine/src/main/resources/org/vividus/bdd/spring.xml:116]: noisy change

File: vividus/src/main/java/org/vividus/bdd/groovy/VariableContextAwareGroovyScriptEvaluator.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * limitations under the License.
+ */
+
+package org.vividus.bdd.groovy;
+
+import org.springframework.scripting.ScriptSource;
+import org.springframework.scripting.groovy.GroovyScriptEvaluator;
+import org.vividus.bdd.context.IBddVariableContext;
+
+public class VariableContextAwareGroovyScriptEvaluator extends GroovyScriptEvaluator
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/bdd/groovy/VariableContextAwareGroovyScriptEvaluator.java:23]: is `GroovyScriptEvaluator` thread-safe? (in other words: is it safe to use it for multiple evaluations and from different threads?)

File: vividus/src/main/resources/org/vividus/spring-vividus.xml [hunk 4]
```diff
@@ -67,6 +67,9 @@
         </property>
     </bean>
 
+
+    <bean id="groovyEvaluator" class="org.vividus.bdd.groovy.VariableContextAwareGroovyScriptEvaluator" />
```
Comment (valfirst) [inline vividus/src/main/resources/org/vividus/spring-vividus.xml:71]: `id` is not needed

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus/src/main/java/org/vividus/bdd/groovy/VariableContextAwareGroovyScriptEvaluator.java (hunk 3 above)
Comment (ikalinin1) [reply vividus/src/main/java/org/vividus/bdd/groovy/VariableContextAwareGroovyScriptEvaluator.java:23]: image
image

Comment (ikalinin1) [reply vividus/src/main/java/org/vividus/bdd/groovy/VariableContextAwareGroovyScriptEvaluator.java:23]: 100 threads

**Review by valfirst — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**